s3datarepository/s3datarepository_test.go

```diff
@@ -0,0 +1,429 @@
+package s3datarepository
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/YaleSpinup/ds-api/apierror"
+	"github.com/YaleSpinup/ds-api/dataset"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3iface"
+)
+
+// mockS3Client is a fake S3 client
+type mockS3Client struct {
+	s3iface.S3API
+	t         *testing.T
+	err       map[string]error
```

Comment:
interesting way to handle mocking more than one call in a tested function, i might steal it 😆 

you could wrap the error too... but this is more straightforward.